### PR TITLE
Menu exports

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Exports the following from `Menu`:
- `MenuDescendantsContext`
  - Used to register and consume Menu descendants
  - This context value allows you to register and track custom menu items
- Context hooks `useMenuContext`, `useMenuGroupContext`, & `useSubMenuContext`, along with types types `MenuContextData`, `MenuGroupContextData`, & `SubMenuContextData`,
  - use these to read context data from custom menu item components
- `menuColor`: Custom color tokens used within `Menu` and related components
- `MenuInteractionState`: Enumerates interaction states on a menu item
- `LGIDs`: Unique ids for menu elements
- `menuItemClassName`: The unique class name for menu item components
- `subMenuContainerClassName` & `subMenuToggleClassName`
  - Unique classnames for submenu elements

Also, adds missing `useHighlightContext` export from `Descendants`

### CODE REVIEW

File: packages/menu/src/MenuItem/MenuItem.types.ts [hunk 1]
```diff
@@ -54,5 +52,5 @@ export interface MenuItemProps {
    * @deprecated - Size no longer has any effect
    */
   // TODO: codemod to remove `size` props from existing implementations
-  size?: Size;
+  size?: 'default' | 'large';
```
Comment (shaneeza) [inline packages/menu/src/MenuItem/MenuItem.types.ts:55]: Why the switch from `Size`?

File: packages/menu/src/testUtils/withMenuContextDecorator.testutils.tsx [hunk 2]
```diff
@@ -15,11 +16,20 @@ import { MenuItem } from '../MenuItem';
...
+        highlight,
+        moveHighlight,
+        setHighlight,
+      },
     } = ctx ?? {
       args: {
         darkMode: false,
         renderDarkMenu: false,
+        highlight: undefined,
+        moveHighlight: noop,
```
Comment (shaneeza) [inline packages/menu/src/testUtils/withMenuContextDecorator.testutils.tsx:31]: What's the diff between `noop` and `undefined`?

**Review by shaneeza — APPROVED** [review]

File: packages/menu/src/testUtils/withMenuContextDecorator.testutils.tsx (hunk 2 above)
Comment (TheSonOfThomp) [reply packages/menu/src/testUtils/withMenuContextDecorator.testutils.tsx:31]: `moveHighlight` has to be defined, so here we use `noop`, which is a function that returns undefined

File: packages/menu/src/MenuItem/MenuItem.types.ts (hunk 1 above)
Comment (TheSonOfThomp) [reply packages/menu/src/MenuItem/MenuItem.types.ts:55]: We've deprecated the `size` prop, and I wanted to remove the type def as well. Having the inline types will avoid breaking TS, but let's us remove old files

**Pull request merged**